src/Reflex/Class.hs

```diff
@@ -791,6 +795,64 @@ zipListWithEvent f l e = do
   mapAccumMaybe_ f' l e
 
 --------------------------------------------------------------------------------
+-- AttachApply
+--------------------------------------------------------------------------------
+
+-- | An 'AttachApply' type carries a function that can be applied to an 'Event' whenever the 'Event' fires.
+--
+-- 'AttachApply' is intended to be used via the '<@>' and '<@' operators and in conjunction with the `Applicative` interface.
+--
+-- If we have:
+-- > f  :: a -> b -> c -> d
+-- > g  :: a -> b -> d
+-- > b1 :: Behavior t a
+-- > b2 :: Behavior t b
+-- > e  :: Event t c
+-- and an 'AttachApply' instance for `Behavior`, then we get:
+-- > f <$> b1 <*> b2 <@> e :: Event t d
+-- > g <$> b1 <*> b2 <@  e :: Event t d
+--
+-- It is a handy way to combine several 'Behavior's or 'Dynamic's together so that you can sample them all at the times when an `Event` is firing.
+class Reflex t => AttachApply t f where
+  attachApply :: f (a -> b) -> Event t a -> Event t b
```

Comment:
From what I can see, this would also make a stronger case for `Reflex t` as a superclass of `AttachApply`, since it's needed to work with `PushM` regardless of which instance we're working with.